### PR TITLE
[AzureMonitorDistro] Integration Tests for Distro

### PR DESCRIPTION
**Pull request by TimothyMothra**
### Goals
We need E2E integration tests to validate that the Distro project can successfully instrument an ASP.NET Core app.

### Changes
- added new Integration Test `VerifyDistro` using Azure SDK's Test Framework.
  This test uses an in-proc WebApplication to test our `UseAzureMonitor` method.
  This test will evaluate all three signals and confirm that expected telemetry is generated and ingested.
- add SessionRecords required by Test Framework
- added a Readme which identifies the steps to get this test running on a local dev box and how to update recordings.

### Future work
For this PR, I'm setting up the overall test scaffolding and simply testing that telemetry exists.
In future PRs, I'll add more column-level validations.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+using Azure.Monitor.Query;
+using Azure.Monitor.Query.Models;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Resources;
+
+#if NET7_0
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:18]: why not NET6?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:18]: Sorry, I added that on day one to stay focused.
I forgot that was still there, will remove it in the final PR.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            // NOTE: If this test starts failing, Flushing may be necessary.
+            //var tracerProvider = app.Services.GetRequiredService<TracerProvider>();
+            //tracerProvider.ForceFlush();
+            //tracerProvider.Shutdown();
+
+            await app.StopAsync(); // shutdown to prevent collecting the log queries.
+
+            // ASSERT
+            // TODO: NEED TO PERFORM COLUMN LEVEL VALIDATIONS.
+            await VerifyLogs(
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:101]: probably good to add why we are querying `AppDependencies` and not `Dependencies` as a note

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+#if NET6_0_OR_GREATER
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests
+{
+    public class DistroWebAppLiveTests : RecordedTestBase<MonitorExporterTestEnvironment>
+    {
+        private const string _testServerUrl = "http://localhost:9998/";
+
+        // DEVELOPER TIP: Change roleName to something unique when working locally (Example "Test##") to easily find your records.
+        // Can search for all records in the portal using this query:   Union * | where AppRoleName == 'Test##'
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:26]: mention LA workspace.

**Review by vishweshbankwar — APPROVED** [review]
LGTM with minor suggestions.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        private LogsQueryClient? _logsQueryClient = null;
+
+        // DEVELOPER TIP: Can pass RecordedTestMode.Live into the base ctor to run this test with a live resource.
+        // DEVELOPER TIP: Can pass RecordedTestMode.Record into the base ctor to re-record the SessionRecords.
+        public DistroWebAppLiveTests(bool isAsync) : base(isAsync) { }
+
+        [RecordedTest]
+        public async Task VerifyDistro()
+        {
+            // SETUP LOGS CLIENT
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:40]: Should we call logs or telemetry. Logs is also a signal, could make it hard to read the code.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:40]: Good point! i'll make the change

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:101]: Are you looking for a note that we're querying using the LogsAnalytics schema, or something else?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs]: Yes - that looks good

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Resources;
+
+#if NET6_0_OR_GREATER
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests
+{
+    public class DistroWebAppLiveTests : RecordedTestBase<MonitorExporterTestEnvironment>
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:21]: Do you want to use this name `MonitorExporterTestEnvironment`?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:21]: That class already exists, but was migrated from the Exporter Integration Tests. I can rename it

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:21]: You could cover in a follow up PR.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+            var builder = WebApplication.CreateBuilder();
+            builder.Logging.ClearProviders();
+            builder.Services.AddOptions<OpenTelemetryLoggerOptions>()
+                .Configure(options =>
+                {
+                    options.SetResourceBuilder(resourceBuilder);
+                });
+            builder.Services.AddOpenTelemetry()
+                .ConfigureResource(x => x.AddAttributes(resourceAttributes))
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:69]: Does this invoke `ResourceBuilder.CreateDefault()` or `ResourceBuilder.CreateEmpty()` with attributes? 

Want to understand as the rolename could be different for trace and request/dependency telemetry.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs (hunk 6 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs:69]: I'm calling `CreateDefault()` on line 58. 

In my Asserts below, I'm querying telemetry by the RoleName to confirm that it's the same for all three signals.

**Pull request merged**